scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/context/TaskContext.java

```diff
@@ -60,29 +60,40 @@
     private final Decrypter decrypter;
     private final TaskResult[] previousTasksResults;
 
-    private final NodeDataSpacesURIs nodeDataSpacesURIs;
+    private final String scratchURI;
+
 
+    private final String inputURI;
+    private final String outputURI;
+    private final String userURI;
+    private final String globalURI;
     private final String schedulerHome;
 
     private final String progressFilePath;
 
 
     public TaskContext(ExecutableContainer executableContainer, TaskLauncherInitializer initializer,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Constructor has 10 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS00107)